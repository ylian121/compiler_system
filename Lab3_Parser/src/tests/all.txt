func main (int a, int b) {
    int x;
    int[5] arr;
    int y = 1;
    arr[4] = 2;
    x = 3;

    while x > y {
        if arr[4] != 0 {
            if x {
                while y+3*2 {
                    print(1 + 5 * call(3, 2 + 4));
                    break;
                }
                continue;
            }
        }
        else {
            read(arr[2]);
            read(x);
        }

    }

    x = (x + 3) * 2 * (3 + 1) % 2;
    x = 1 <= 2 >= 2 * 3 + 2*4 + 1;
    x = 1 < 2 == 1 > 2 != 1 < 2;

    return arr[0] + arr[1];    
}


