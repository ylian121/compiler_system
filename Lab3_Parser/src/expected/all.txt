function header: main, a, b
{
declare var: x
declare array: arr, 5
declare var: y
assign var: y = 1
assign arr: arr[4] = 2
assign var: x = 3
while(temp0 = x > y
 cond(temp0))
{
if(assign: temp1 = arr[4]
temp2 = temp1 != 0
 cond(temp2))
{
if( cond(x))
{
while(temp3 = 3 * 2
temp4 = y + temp3
 cond(temp4))
{
temp5 = 2 + 4
call: temp6 = call(3, temp5, )
temp7 = 5 * temp6
temp8 = 1 + temp7
print: temp8
break
}
continue
}
}
else
{
read: arr[2]
read: x
}
}
temp9 = x + 3
temp10 = temp9 * 2
temp11 = 3 + 1
temp12 = temp10 * temp11
temp13 = temp12 % 2
assign var: x = temp13
temp14 = 1 <= 2
temp15 = 2 * 3
temp16 = 2 * 4
temp17 = temp15 + temp16
temp18 = temp17 + 1
temp19 = temp14 >= temp18
assign var: x = temp19
temp20 = 1 < 2
temp21 = 1 > 2
temp22 == temp20 > temp21
temp23 = 1 < 2
temp24 = temp22 != temp23
assign var: x = temp24
assign: temp25 = arr[0]
assign: temp26 = arr[1]
temp27 = temp25 + temp26
return: temp27
}
